src/main/kotlin/org/rust/ide/inspections/duplicates/RustDuplicateStructFieldInspection.kt

```diff
@@ -2,21 +2,18 @@ package org.rust.ide.inspections.duplicates
 
 import com.intellij.codeInspection.ProblemHighlightType
 import com.intellij.codeInspection.ProblemsHolder
-import com.intellij.psi.PsiElement
 import org.rust.ide.inspections.RustLocalInspectionTool
-import org.rust.lang.core.psi.RustElementVisitor
+import org.rust.lang.core.psi.RustFieldDeclElement
 import org.rust.lang.core.psi.RustFieldsOwner
 import org.rust.lang.core.psi.fields
 
 class RustDuplicateStructFieldInspection : RustLocalInspectionTool() {
-    override fun buildVisitor(holder: ProblemsHolder, isOnTheFly: Boolean) = object : RustElementVisitor() {
-        override fun visitElement(element: PsiElement) {
-            if (element is RustFieldsOwner) {
-                for (dupe in element.fields.findDuplicates()) {
-                    holder.registerProblem(dupe, "Duplicate field <code>#ref</code>", ProblemHighlightType.GENERIC_ERROR_OR_WARNING)
-                }
+    override fun buildVisitor(holder: ProblemsHolder, isOnTheFly: Boolean) =
+            RustDuplicateItem.createInspection(RustFieldsOwner::getFields) {
+                holder.registerProblem(it, "Duplicate field <code>#ref</code>", ProblemHighlightType.GENERIC_ERROR_OR_WARNING)
             }
-        }
-    }
 }
 
+//TODO: Why do we need this extra layer of indirection? Why can't we rely on the property's getter via reference?
+private fun RustFieldsOwner.getFields(): List<RustFieldDeclElement> = fields
```

Comment:
Yeah, this was probably the single most confusing part of this PR.

IntelliJ seems to be okay with letting you pass `RustFieldsOwner::fields` to `createInspection()`, but the compiler doesn't like that. I really only figured it out after taking a look at the generated bytecode :(